spec/library_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'yaml'
+require './lib/library.rb'
+
+collection = YAML.load_file('./lib/data.yml')
+describe Library do
+
+  subject { described_class.new(name: 'Fabian') }
+
+  it 'list all books in library' do
+    expect(subject.all_books).not_to eq nil
+  end
```

Comment:
Ok very good. Just keep in mind you have two things going on - `collection` and `@books`. Do you need both?